### PR TITLE
feat(SMTM-8): list marketplace items

### PR DESCRIPTION
**Pull request by FilippoRezzonico**
[SMTM-8](https://makeitapp.atlassian.net/browse/SMTM-8)

### CODE REVIEW

File: internal/cmd/marketplace/list.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+func listMarketplaceItems(client *client.APIClient, companyID string) error {
+	req := client.
+		Get().
+		APIPath(listMarketplaceEndpoint)
+
+	if len(companyID) != 0 {
+		req.SetParam("tenantId", companyID)
+	}
+	// execute the request
+	resp, err := req.Do(context.Background())
```
Comment (fredmaggiowski) [inline internal/cmd/marketplace/list.go:62]: I'd be better maybe to split the API invocation from the "table rendering" in two functions, this way we could re-use the list api in the future

File: internal/cmd/marketplace.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"github.com/mia-platform/miactl/internal/clioptions"
+	"github.com/mia-platform/miactl/internal/cmd/marketplace"
+	"github.com/spf13/cobra"
+)
+
+func MarketplaceCmd(options *clioptions.CLIOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "marketplace",
+		Short: "view and manage mia marketplace items",
```
[the remark this answers is not in the document]
Comment (FilippoRezzonico) [reply internal/cmd/marketplace.go]: "mia" has been removed now

File: internal/cmd/marketplace/list.go (hunk 1 above)
Comment (FilippoRezzonico) [reply internal/cmd/marketplace/list.go]: Now the API invocation is in a dedicated function 👍🏼

File: internal/cmd/marketplace/list.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	}
+
+	return marketplaceItems, nil
+}
+
+// listMarketplaceItems retrieves the marketplace items belonging to the current context
+func listMarketplaceItems(client *client.APIClient, companyID string) error {
+	marketplaceItems, err := getMarketplaceItemsByCompany(client, companyID)
+	if err != nil {
+		return err
```
Comment (malta895) [inline internal/cmd/marketplace/list.go:82]:
```suggestion
		return fmt.Errorf( "error getting marketplace items: %w, err)
```

File: internal/cmd/marketplace.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"github.com/mia-platform/miactl/internal/clioptions"
+	"github.com/mia-platform/miactl/internal/cmd/marketplace"
+	"github.com/spf13/cobra"
+)
+
+func MarketplaceCmd(options *clioptions.CLIOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "marketplace",
+		Short: "view and manage marketplace items",
```
[the remark this answers is not in the document]
Comment (FilippoRezzonico) [reply internal/cmd/marketplace.go]: Applicato

File: internal/cmd/marketplace.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	}
+
+	// add cmd flags
+	flags := cmd.PersistentFlags()
+	options.AddConnectionFlags(flags)
+	options.AddCompanyFlags(flags)
+	options.AddContextFlags(flags)
+
+	// add sub commnads
```
[the remark this answers is not in the document]
Comment (FilippoRezzonico) [reply internal/cmd/marketplace.go]: Applicato

File: internal/cmd/marketplace/list.go [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+		},
+	}
+}
+
+func getMarketplaceItemsByCompany(client *client.APIClient, companyID string) ([]*resources.MarketplaceItem, error) {
+	if len(companyID) == 0 {
+		return nil, fmt.Errorf("missing company id, please set one with the flag or context")
+	}
+
+	// execute the request
```
[the remark this answers is not in the document]
Comment (FilippoRezzonico) [reply internal/cmd/marketplace/list.go]: Rimosso

File: internal/cmd/marketplace/list.go [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+			restConfig, err := options.ToRESTConfig()
+			cobra.CheckErr(err)
+			client, err := client.APIClientForConfig(restConfig)
+			cobra.CheckErr(err)
+			return listMarketplaceItems(client, restConfig.CompanyID)
+		},
+	}
+}
+
+func getMarketplaceItemsByCompany(client *client.APIClient, companyID string) ([]*resources.MarketplaceItem, error) {
```
Comment (pmarchini) [inline internal/cmd/marketplace/list.go:50]:
```suggestion
func getMarketplaceItemsByCompanyId(client *client.APIClient, companyID string) ([]*resources.MarketplaceItem, error) {
```

File: internal/cmd/marketplace/list.go [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+	marketplaceItems := make([]*resources.MarketplaceItem, 0)
+	if err := resp.ParseResponse(&marketplaceItems); err != nil {
+		return nil, fmt.Errorf("error parsing response body: %w", err)
+	}
+
+	return marketplaceItems, nil
+}
+
+// listMarketplaceItems retrieves the marketplace items belonging to the current context
```
Comment (pmarchini) [inline internal/cmd/marketplace/list.go:77]: Is this comment still valid?

File: internal/cmd/marketplace/list_test.go [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+		},
+		"repositoryUrl": "https://git.tools.mia-platform.eu/platform/core/acl-service",
+		"componentsIds": [],
+		"publishOnMiaDocumentation": true,
+		"documentation": {
+			"type": "externalLink",
+			"url": "https://docs.mia-platform.eu/docs/runtime_suite/acl-service/overview"
+		}
+	}
+]`
```
Comment (pmarchini) [inline internal/cmd/marketplace/list_test.go:100]: I suggest replacing the content with something unrelated to the product, perhaps something humorous or entertaining 🚀.

File: internal/cmd/marketplace/list.go (hunk 8 above)
Comment (FilippoRezzonico) [reply internal/cmd/marketplace/list.go:77]: yes it is

File: internal/cmd/marketplace/list_test.go (hunk 9 above)
Comment (FilippoRezzonico) [reply internal/cmd/marketplace/list_test.go]: I replaced the ACL Service with the "Space Travel Service" 😎

**Review by pmarchini — APPROVED** [review]

File: internal/cmd/marketplace/list_test.go [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+		"description": "This service provides a REST API to book your next journey to space!",
+		"type": "plugin",
+		"imageUrl": "/v2/files/download/space.png",
+		"supportedByImageUrl": "/v2/files/download/23b12dd9-43a6-4920-cb2d-2a809d946851.png",
+		"supportedBy": "My-Company",
+		"category": {
+			"id": "auth",
+			"label": "Core Plugins - Travel"
+		},
+		"repositoryUrl": "https://git.tools.my-company.eu/plugins/core/space-travel-service",
```
Comment (fredmaggiowski) [inline internal/cmd/marketplace/list_test.go:92]:
```suggestion
		"repositoryUrl": "https://git.com/plugins/core/space-travel-service",
```

**Review by davidebianchi — APPROVED** [review]

**Pull request merged**